vendor/github.com/aead/poly1305/poly1305_ref.go

```diff
@@ -0,0 +1,228 @@
+// Copyright (c) 2016 Andreas Auernhammer. All rights reserved.
+// Use of this source code is governed by a license that can be
+// found in the LICENSE file.
+
+// +build !amd64 gccgo appengine nacl
+
+package poly1305
+
+import "encoding/binary"
+
+const (
+	msgBlock   = uint32(1 << 24)
+	finalBlock = uint32(0)
+)
+
+// Sum generates an authenticator for msg using a one-time key and returns the
+// 16-byte result. Authenticating two different messages with the same key allows
+// an attacker to forge messages at will.
+func Sum(msg []byte, key [32]byte) [TagSize]byte {
+	var (
+		h, r  [5]uint32
+		s     [4]uint32
+		block [TagSize]byte
```

Comment:
It is not necessarily better (compiler-wise), maybe a little neater to allocate more closely to where you need it (and bound to scope).